aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move

```diff
@@ -7,6 +7,8 @@ module aptos_framework::optional_aggregator {
     use aptos_framework::aggregator_factory;
     use aptos_framework::aggregator::{Self, Aggregator};
 
+    friend aptos_framework::coin;
```

Comment:
What happens if we want to open this up for public usages in the future?

Reply:
Good question @movekevin . For supply - we upgrade. Everything else is tricky since we really don't want to expose it yet. Maybe have one more `create_optional_aggregator` method that takes a signer as well?

Reply:
Okay that makes sense. So in the future when we want to open this up, we'll just add a non-friend public function? That can work also.

Reply:
Yes, we ll just add a non-friend. Re signer - I will add that as well with a check if it's aptos account (and the implementation can be upgraded to drop that check)